### PR TITLE
Some minor fixes / improvements

### PR DESCRIPTION
**Pull request by octonato**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: "3"
 services:
   akka-serverless-proxy:
-    image: gcr.io/akkaserverless-public/akkaserverless-proxy:0.8.2
+    image: gcr.io/akkaserverless-public/akkaserverless-proxy:0.8.13
```
Comment (octonato) [inline docker-compose.yml:4]: Some queries were not supported by 0.8.2. I guess you have another version running locally? 

Anyway, 0.8.13 is the most recent one. I do get some warnings in the logs, but didn't spend time digging in. Probably something that we can see when moving to Kalix.

File: src/main/java/io/mystore/cart/entity/ShoppingCart.java [hunk 2]
```diff
@@ -342,8 +342,8 @@ static CartEntity.CartState updateState(CartEntity.CartState state, CartEntity.C
   static List<CartEntity.LineItem> updateLineItemQuantity(CartEntity.CartState state, CartEntity.ItemAdded event) {
     return state.getLineItemsList().stream()
         .map(lineItem -> {
-          if (lineItem.getSkuId().equals(lineItem.getSkuId())) {
-            return lineItem.toBuilder()
+          if (lineItem.getSkuId().equals(event.getLineItem().getSkuId())) {
```
Comment (octonato) [inline src/main/java/io/mystore/cart/entity/ShoppingCart.java:345]: This was a bug. It was returns true and updates all times with the new quantity.

(also below in line 359)

File: src/main/java/io/mystore/shipping/action/ShippingToOrderSkuItemAction.java [hunk 3]
```diff
@@ -36,9 +36,9 @@ public Effect<Empty> onOrderCreated(ShippingEntity.OrderCreated orderCreated) {
         .toList();
 
     var result = CompletableFuture.allOf(results.toArray(new CompletableFuture[results.size()]))
-        .thenApply(reply -> effects().reply(Empty.getDefaultInstance()));
+        .thenApply(reply -> Empty.getDefaultInstance());
 
-    return effects().asyncEffect(result);
+    return effects().asyncReply(result);
```
Comment (octonato) [inline src/main/java/io/mystore/shipping/action/ShippingToOrderSkuItemAction.java:41]: It's not needed to map to `Effect<Empty>`. We can return a `CompletionStage<Empty>` using `asyncReply`.

**Comment by mckeeh3** [comment]
Thanks for the detailed code review and detailed fixes.

**Pull request merged**